### PR TITLE
[stablehlo] Update stablehlo to 341e063f0924fc1350538dc53a92c21ec5e022a3

### PR DESCRIPTION
**Pull request by bviyer**
Brought StableHLO to its latest HEAD.

Major change in this checkin: Unranked dynamism was removed from StableHLO - https://github.com/openxla/stablehlo/blob/main/rfcs/20230704-dynamism-101.md#-p5-represent-shape-computations-as-stablehlo-operations-on-variadic-0-dimensional-tensors-and-drop-support-for-unranked-dynamism

### CODE REVIEW

File: compiler/plugins/input/StableHLO/Conversion/Preprocessing/test/complex_lowering.mlir [hunk 1]
```diff
@@ -15,21 +15,6 @@ func.func @add(%arg0 : tensor<2xf32>, %arg1 : tensor<2xf32>, %arg2 : tensor<2xf3
   func.return %5, %6 : tensor<2xf32>, tensor<2xf32>
 }
 
-// CHECK-LABEL: @add_unranked
-func.func @add_unranked(%arg0 : tensor<*xf32>, %arg1 : tensor<*xf32>, %arg2 : tensor<*xf32>, %arg3 : tensor<*xf32>) -> (tensor<*xf32>, tensor<*xf32>) {
```
Comment (ScottTodd) [inline compiler/plugins/input/StableHLO/Conversion/Preprocessing/test/complex_lowering.mlir:19]: Can you add a note to the PR description about why these unranked tests are removed? Maybe link to a relevant commit / email from stablehlo if unranked is no longer supported (I don't think IREE handles unranked anyways)

**Review by ScottTodd — APPROVED** [review]

File: compiler/plugins/input/StableHLO/Conversion/Preprocessing/test/complex_lowering.mlir (hunk 1 above)
Comment (bviyer) [reply compiler/plugins/input/StableHLO/Conversion/Preprocessing/test/complex_lowering.mlir:19]: There are unranked tests in multiple places. Do you think its good to add there? I was thinking of adding it in the commit log. Something like this:

```

Major change in this checkin:

1. Unranked dynamism was removed from StableHLO - https://github.com/openxla/stablehlo/blob/main/rfcs/20230704-dynamism-101.md#-p5-represent-shape-computations-as-stablehlo-operations-on-variadic-0-dimensional-tensors-and-drop-support-for-unranked-dynamism
```

Comment (ScottTodd) [reply compiler/plugins/input/StableHLO/Conversion/Preprocessing/test/complex_lowering.mlir:19]: You can include it in both the PR description and the commit log.

We have GitHub configured to use the PR description as the commit message when merging by default: 
image

(you can change the commit message if you'd like, and I do sometimes, but keeping information in both places is useful)

Comment (ScottTodd) [reply compiler/plugins/input/StableHLO/Conversion/Preprocessing/test/complex_lowering.mlir:19]: Oh, and use a public link for that, `http://go/` only works for Google :P

**Review by ScottTodd — APPROVED** [review]

File: compiler/plugins/input/StableHLO/Conversion/Preprocessing/test/complex_lowering.mlir (hunk 1 above)
Comment (bviyer) [reply compiler/plugins/input/StableHLO/Conversion/Preprocessing/test/complex_lowering.mlir:19]: whoops 🤦🏽

**Review by julianwa — APPROVED** [review]

**Review by rsuderman — APPROVED** [review]

**Pull request merged**